src/main/kotlin/org/dashj/platform/dpp/statetransition/StateTransition.kt

```diff
@@ -101,8 +111,26 @@ abstract class StateTransition(
         return encodeProtocolEntity(serializedData)
     }
 
-    fun signByPrivateKey(privateKey: String) {
-        signByPrivateKey(ECKey.fromPrivate(Converters.fromHex(privateKey)))
+    fun signByPrivateKey(privateKey: String, keyType: IdentityPublicKey.Type) {
+        signByPrivateKey(Converters.byteArrayFromString(privateKey), keyType)
+    }
+    fun signByPrivateKey(privateKey: ByteArray, keyType: IdentityPublicKey.Type) {
+        val data = hash(skipSignature = true)
+
+        when (keyType) {
+            IdentityPublicKey.Type.ECDSA_SECP256K1,
+            IdentityPublicKey.Type.ECDSA_HASH160 -> {
+                val privateKeyModel = ECKey.fromPrivate(privateKey)
+                signature = privateKeyModel.signHash(Sha256Hash.wrap(data))
+            }
+            IdentityPublicKey.Type.BLS12_381 -> {
+                val privateKeyModel = BLSSecretKey(privateKey)
+                val blsSignature = privateKeyModel.Sign(Sha256Hash.wrap(data))
+
+                signature = blsSignature.bitcoinSerialize()
+            }
```

Comment:
One big change is adding support for BLS public keys in the identity for digital signatures (sign / verify)

Reply:
DashPay won't be supporting BLS in identities for the time being.